packages/talos_agent/src/agentv2/agent_v2.rs

```diff
@@ -0,0 +1,58 @@
+use crate::agentv2::model::{CancelRequestChannelMessage, CertifyRequestChannelMessage};
+use crate::agentv2::state_manager::StateManager;
+use crate::api::{AgentConfig, CertificationRequest, CertificationResponse, KafkaConfig, TalosAgent, TalosIntegrationType};
+use async_trait::async_trait;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use time::OffsetDateTime;
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::{Receiver, Sender};
+
+pub struct TalosAgentImplV2 {
+    rx_certify: Sender<CertifyRequestChannelMessage>,
+    _rx_cancel: Sender<CancelRequestChannelMessage>,
+}
+
+impl TalosAgentImplV2 {
+    pub async fn new(
+        agent_config: AgentConfig,
+        kafka_config: Option<KafkaConfig>,
+        int_type: &TalosIntegrationType,
+        certify: (Sender<CertifyRequestChannelMessage>, Receiver<CertifyRequestChannelMessage>),
+        cancel: (Sender<CancelRequestChannelMessage>, Receiver<CancelRequestChannelMessage>),
+        publish_times: &Arc<Mutex<HashMap<String, u64>>>,
+    ) -> Result<TalosAgentImplV2, String> {
+        let state_manager = StateManager::new(agent_config, kafka_config, int_type, publish_times).unwrap();
+
+        state_manager.start(certify.1, cancel.1).await;
```

Comment:
This shouldn't be done inside the `new` function. Ideally we only expect a new instance of `Self/Struct` to be created here.

Reply:
Thanks. This is fixed now.